src/ba.py

```diff
@@ -2,36 +2,63 @@
 
 
 def border_array(x: str) -> list[int]:
-    """
-    Construct the border array for x.
+    #edge case
+    if x == "":
+        return []
+    #built border array
+    ba = [0]*len(x)
+    
+    #otherwise run trough seq
+    for i in range(1, len(x)):
+        b = ba[i-1]
+        #extend border
+        while (b > 0) and (x[i] != x[b]):
+            b = ba[b-1]
+        #if matches
+        if(x[i]==x[b]):
+            ba[i] = b + 1
+        else:
+            ba[i] = 0
 
-    >>> border_array("aaba")
-    [0, 1, 0, 1]
-    >>> border_array("ississippi")
-    [0, 0, 0, 1, 2, 3, 4, 0, 0, 1]
-    >>> border_array("")
-    []
-    """
-    return []  # FIXME
+    return ba
 
 
 def strict_border_array(x: str) -> list[int]:
-    """
-    Construct the strict border array for x.
-
-    A struct border array is one where the border cannot
-    match on the next character. If b is the length of the
-    longest border for x[:i+1], it means x[:b] == x[i-b:i+1],
-    but for a strict border, it must be the longest border
-    such that x[b] != x[i+1].
-
-    >>> strict_border_array("aaba")
-    [0, 1, 0, 1]
-    >>> strict_border_array("aaaba")
-    [0, 0, 2, 0, 1]
-    >>> strict_border_array("ississippi")
-    [0, 0, 0, 0, 0, 0, 4, 0, 0, 1]
-    >>> strict_border_array("")
-    []
-    """
-    return []  # FIXME
+    #edge case
+    if x == "":
+        return []
+        
+    ba = border_array(x)
+    #built border array
+    bax = [0]*len(x)
+    
+    #otherwise run trough seq
+    for i in range(len(x)-1):
+        if (ba[i] != 0 and x[ba[i]] != x[i+1]):
+            bax[i] = ba[i]
+        else:
+            bax[i] = 0
```

Comment:
If the border isn't strict, it doesn't mean that there is no strict border. There could be a shorter borer that is also strict.

Consider the string `abaabaa` and the second-to-last index. The border is `aba` but it isn't strict, since both `x[:ba[i]]` and `x[:i]` is followed by an `a`.

```
aba a ba a
aba[a]
    a ba[a]
```

But the strict border isn't empty either. It is just shorter. The string `a` is both a prefix and suffix of `abaaba` and the border is followed by `b` while `x[:i]` is followed by `a`, so that is a strict border.

```
ab aaba a
a[b]
        a[a]
```